In [4]:
import pandas as pd
df05=pd.read_csv("data/05-Internet_Accesos-por-tecnologia.csv")
df13=pd.read_csv("data/13-Internet_Ingresos.csv")


## En este notebook la idea es juntar las dos tablas con información por trimestre y por tipo de conectividad (ADSL,Cablemodem, Fibra óptica, Wireless) , a los fines de obtener ingresos discriminados por tipo de conectividad 

In [12]:
df05

Año  Trimestre  ADSL  Cablemodem  Fibra óptica  Wireless  Otros  Total   
0   2022          4     1           6             3       573    222     11  \
1   2022          3     1           6             3       557    235     11   
2   2022          2     1           6             3       556    232     11   
3   2022          1     2           6             2       546    252     11   
4   2021          4     2           6             2       523    253     10   
5   2021          3     2           6             2       492    250     10   
6   2021          2     2           6             1       477    254     10   
7   2021          1     2           5             1       435    250     10   
8   2020          4     2           5             1       422    253     10   
9   2020          3     2           5             1       413    249      9   
10  2020          2     2           5             1       377    247      9   
11  2020          1     2           5             1       352    209      9   
12  2019          4     2           5           941       340    213      9   
13  2019          3     3           5           852       338    264      9   
14  2019          2     3           5           761       308    265      9   
15  2019          1     3           4           637       276    166      9   
16  2018          4     3           4           388       264    178      8   
17  2018          3     4           4           247       246     94      8   
18  2018          2     4           4           226       194     95      8   
19  2018          1     4           4           217       165     99      8   
20  2017          4     4           4           211       156    101      8   
21  2017          3     4           4           194       106    132      8   
22  2017          2     4           3           181        86     56      7   
23  2017          1     4           3           180        85     56      7   
24  2016          4     4           3           178        85     55      7   
25  2016          3     4           3           170        81     54      7   
26  2016          2     4           3           168        85     59      7   
27  2016          1     4           3           164        85     59      7   
28  2015          4     4           3           139        86     59      7   
29  2015          3     4           3           163        85     59      7   
30  2015          2     4           3           151        82     59      7   
31  2015          1     4           3           168        79     66      7   
32  2014          4     4           3           150        77     72      7   
33  2014          3     4           3           155        85     70      7   
34  2014          2     4           2           149        72     72      6   
35  2014          1     4           2           150        71     73      6   

          Periodo  
0    Oct-Dic 2022  
1   Jul-Sept 2022  
2    Abr-Jun 2022  
3    Ene-Mar 2022  
4    Oct-Dic 2021  
5   Jul-Sept 2021  
6    Abr-Jun 2021  
7    Ene-Mar 2021  
8    Oct-Dic 2020  
9   Jul-Sept 2020  
10   Abr-Jun 2020  
11   Ene-Mar 2020  
12   Oct-Dic 2019  
13  Jul-Sept 2019  
14   Abr-Jun 2019  
15   Ene-Mar 2019  
16   Oct-Dic 2018  
17  Jul-Sept 2018  
18   Abr-Jun 2018  
19   Ene-Mar 2018  
20   Oct-Dic 2017  
21  Jul-Sept 2017  
22   Abr-Jun 2017  
23   Ene-Mar 2017  
24   Oct-Dic 2016  
25  Jul-Sept 2016  
26   Abr-Jun 2016  
27   Ene-Mar 2016  
28   Oct-Dic 2015  
29  Jul-Sept 2015  
30   Abr-Jun 2015  
31   Ene-Mar 2015  
32   Oct-Dic 2014  
33  Jul-Sept 2014  
34   Abr-Jun 2014  
35   Ene-Mar 2014

No se entiende la variabilidad de los datos que pasan de 941 a 1 

In [6]:
df13

Año  Trimestre  Ingresos (miles de pesos)        Periodo
0   2022          4                         72   Oct-Dic 2022
1   2022          3                         64  Jul-Sept 2022
2   2022          2                         60   Abr-Jun 2022
3   2022          1                         55   Ene-Mar 2022
4   2021          4                         45   Oct-Dic 2021
5   2021          3                         43  Jul-Sept 2021
6   2021          2                         38   Abr-Jun 2021
7   2021          1                         37   Ene-Mar 2021
8   2020          4                         34   Oct-Dic 2020
9   2020          3                         32  Jul-Sept 2020
10  2020          2                         32   Abr-Jun 2020
11  2020          1                         30   Ene-Mar 2020
12  2019          4                         24   Oct-Dic 2019
13  2019          3                         22  Jul-Sept 2019
14  2019          2                         21   Abr-Jun 2019
15  2019          1                         20   Ene-Mar 2019
16  2018          4                         17   Oct-Dic 2018
17  2018          3                         15  Jul-Sept 2018
18  2018          2                         14   Abr-Jun 2018
19  2018          1                         13   Ene-Mar 2018
20  2017          4                         11   Oct-Dic 2017
21  2017          3                         10  Jul-Sept 2017
22  2017          2                         10   Abr-Jun 2017
23  2017          1                          9   Ene-Mar 2017
24  2016          4                          7   Oct-Dic 2016
25  2016          3                          7  Jul-Sept 2016
26  2016          2                          7   Abr-Jun 2016
27  2016          1                          6   Ene-Mar 2016
28  2015          4                          5   Oct-Dic 2015
29  2015          3                          5  Jul-Sept 2015
30  2015          2                          5   Abr-Jun 2015
31  2015          1                          5   Ene-Mar 2015
32  2014          4                          4   Oct-Dic 2014
33  2014          3                          3  Jul-Sept 2014
34  2014          2                          3   Abr-Jun 2014
35  2014          1                          3   Ene-Mar 2014

Realizo un merge de las 2 tablas para evaluar ingresos y accesos

In [7]:
# Realizar el merge por las columnas 'Año' y 'Trimestre'
df_merged = pd.merge(df05, df13, on=['Año', 'Trimestre'], how='inner')


In [9]:
df_merged.columns

Index(['Año', 'Trimestre', 'ADSL', 'Cablemodem', 'Fibra óptica', 'Wireless',
       'Otros', 'Total', 'Periodo_x', 'Ingresos (miles de pesos)',
       'Periodo_y'],
      dtype='object')

Voy a multiplicar los valores por mil de aquellos menores a 10 porque es obvio que hay un salto de valores,
por ejemplo passa de 949 a 1 en vez de pasar a mil


In [16]:
# Multiplicar las columnas por mil si su valor es menor a 10
# Lista de columnas a multiplicar
columnas_a_multiplicar = ['ADSL', 'Cablemodem', 'Fibra óptica', 'Wireless', 'Otros']

# Multiplicar las columnas por mil si su valor es menor a 10
df_merged[columnas_a_multiplicar] = df_merged[columnas_a_multiplicar].applymap(lambda x: x * 1000 if x < 10 else x)


In [19]:
del df_merged["Periodo_x"]

In [22]:
# Renombrar la columna 'periodo_x' a 'periodo'
df_merged.rename(columns={'Periodo_y': 'Periodo'}, inplace=True)

In [24]:
df_merged.to_csv("data/Accesos_ingresos.csv",index=False)

In [23]:
df_merged

Año  Trimestre  ADSL  Cablemodem  Fibra óptica  Wireless  Otros  Total   
0   2022          4  1000        6000          3000       573    222     11  \
1   2022          3  1000        6000          3000       557    235     11   
2   2022          2  1000        6000          3000       556    232     11   
3   2022          1  2000        6000          2000       546    252     11   
4   2021          4  2000        6000          2000       523    253     10   
5   2021          3  2000        6000          2000       492    250     10   
6   2021          2  2000        6000          1000       477    254     10   
7   2021          1  2000        5000          1000       435    250     10   
8   2020          4  2000        5000          1000       422    253     10   
9   2020          3  2000        5000          1000       413    249      9   
10  2020          2  2000        5000          1000       377    247      9   
11  2020          1  2000        5000          1000       352    209      9   
12  2019          4  2000        5000           941       340    213      9   
13  2019          3  3000        5000           852       338    264      9   
14  2019          2  3000        5000           761       308    265      9   
15  2019          1  3000        4000           637       276    166      9   
16  2018          4  3000        4000           388       264    178      8   
17  2018          3  4000        4000           247       246     94      8   
18  2018          2  4000        4000           226       194     95      8   
19  2018          1  4000        4000           217       165     99      8   
20  2017          4  4000        4000           211       156    101      8   
21  2017          3  4000        4000           194       106    132      8   
22  2017          2  4000        3000           181        86     56      7   
23  2017          1  4000        3000           180        85     56      7   
24  2016          4  4000        3000           178        85     55      7   
25  2016          3  4000        3000           170        81     54      7   
26  2016          2  4000        3000           168        85     59      7   
27  2016          1  4000        3000           164        85     59      7   
28  2015          4  4000        3000           139        86     59      7   
29  2015          3  4000        3000           163        85     59      7   
30  2015          2  4000        3000           151        82     59      7   
31  2015          1  4000        3000           168        79     66      7   
32  2014          4  4000        3000           150        77     72      7   
33  2014          3  4000        3000           155        85     70      7   
34  2014          2  4000        2000           149        72     72      6   
35  2014          1  4000        2000           150        71     73      6   

    Ingresos (miles de pesos)        Periodo  
0                          72   Oct-Dic 2022  
1                          64  Jul-Sept 2022  
2                          60   Abr-Jun 2022  
3                          55   Ene-Mar 2022  
4                          45   Oct-Dic 2021  
5                          43  Jul-Sept 2021  
6                          38   Abr-Jun 2021  
7                          37   Ene-Mar 2021  
8                          34   Oct-Dic 2020  
9                          32  Jul-Sept 2020  
10                         32   Abr-Jun 2020  
11                         30   Ene-Mar 2020  
12                         24   Oct-Dic 2019  
13                         22  Jul-Sept 2019  
14                         21   Abr-Jun 2019  
15                         20   Ene-Mar 2019  
16                         17   Oct-Dic 2018  
17                         15  Jul-Sept 2018  
18                         14   Abr-Jun 2018  
19                         13   Ene-Mar 2018  
20                         11   Oct-Dic 2017  
21                         10  Jul-Sept 2017  
